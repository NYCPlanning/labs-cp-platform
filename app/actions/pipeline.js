import * as AT from '../constants/actionTypes';
import * as cartoActions from './carto';
import { sqlConfig, getSql } from '../helpers/sqlbuilder/PipelineSqlBuilder';

export const fetchDetails = cartodbId =>
  cartoActions.getFeature({
    tableName: sqlConfig.tablename,
    column: 'cartodb_id',
    value: cartodbId,
  }, AT.FETCH_PIPELINE_DETAILS);

export const fetchTotalCount = () => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT COUNT(*)
    FROM ${sqlConfig.tablename}
    WHERE
      the_geom IS NOT NULL
      AND
      u_net IS NOT NULL
      AND
      (
        dcp_status = 'Complete'
        OR dcp_status = 'Partial complete'
        OR dcp_status = 'Permit issued'
        OR dcp_status = 'Application filed'
        OR dcp_status = 'Complete (demolition)'
      )
      AND
      (
        dcp_dev_category = 'New Building'
        OR dcp_dev_category = 'Alteration'
        OR dcp_dev_category = 'Demolition'
      )
      AND
      (
        dcp_occ_category = 'Residential'
        OR dcp_occ_category = 'Other Accommodations'
      )
      AND x_outlier <> 'true'
      AND x_dup_flag = ''`,
    requestFormat: 'json',
    nextType: AT.FETCH_PIPELINE_TOTAL_COUNT,
  },
});

export const fetchSelectedCount = filterDimensions => ({
  type: AT.CARTO_REQUEST,
  payload: {
    sql: `SELECT count(*) FROM (${getSql(filterDimensions)}) a`,
    requestFormat: 'json',
    nextType: AT.FETCH_PIPELINE_SELECTED_COUNT,
  },
});

export const setSelectedFeatures = selectedFeatures => ({
  type: AT.SET_SELECTED_PIPELINE_FEATURES,
  payload: { selectedFeatures },
});

export const setFilterDimension = (filterDimension, values) => ({
  type: AT.SET_PIPELINE_FILTER_DIMENSION,
  payload: { filterDimension, values },
});

export const setSymbology = symbologyDimension => ({
  type: AT.SET_PIPELINE_SYMBOLOGY,
  payload: { symbologyDimension },
});

export const resetFilter = () => ({
  type: AT.RESET_PIPELINE_FILTER,
});
