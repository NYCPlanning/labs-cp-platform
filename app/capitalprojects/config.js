// config.js - various data objects for the multiselectors on the capital projects explorer

var config = {
  agencies: [
      {  
         "value":"ACS",
         "label":"Administration for Children's Services - ACS"
      },
      {  
         "value":"BPL",
         "label":"Brooklyn Public Library - BPL"
      },
      {  
         "value":"CUNY",
         "label":"City University of New York - CUNY"
      },
      {  
         "value":"DCAS",
         "label":"Department of Citywide Administrative Services - DCAS"
      },
      {  
         "value":"DCLA",
         "label":"Department of Cultural Affairs - DCLA"
      },
      {  
         "value":"DDC",
         "label":"Department of Design and Construction - DDC"
      },
      {  
         "value":"DEP",
         "label":"Department of Environmental Protection - DEP"
      },
      {
         "value":"DFTA",
         "label":"Department for the Aging"
      },
      {
         "value":"DHP",
         "label":"Department of Housing Preservation and Development"
      },
      {
         "value":"DHS",
         "label":"Department of Homeless Services"
      },
      {
         "value":"DOC",
         "label":"Department of Correction",
      },
      {  
         "value":"DOHMH",
         "label":"Department of Health and Mental Hygiene - DOHMH"
      },
      {  
         "value":"DOITT",
         "label":"Department of Info Tech and Telecom"
      },
      {  
         "value":"DOT",
         "label":"Department of Transportation - DOT"
      },
      {  
         "value":"DPR",
         "label":"Department of Parks and Recreation - DPR"
      },
      {  
         "value":"DSNY",
         "label":"Department of Sanitation - DNSY"
      },
      {  
         "value":"FDNY",
         "label":"Fire Department - FDNY"
      },
      {  
         "value":"HHC",
         "label":"Health and Hospitals Corporation - HHC"
      },
      {  
         "value":"NYPD",
         "label":"Police Department - NYPD"
      },
      {  
         "value":"NYPL",
         "label":"New York Public Library - NYPL"
      },
      {  
         "value":"NYRL",
         "label":"New York Research Library - NYRL"
      },
      {  
         "value":"QBPL",
         "label":"Queens Borough Public Library - QBPL"
      },
      {  
         "value":"SBS",
         "label":"Department of Small Business Services - SBS"
      },
      {
         "value":"UK",
         "label":"Unknown Agency Code"
      }
   ]
}

module.exports=config