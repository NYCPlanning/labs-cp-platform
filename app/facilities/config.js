import defaultLayers from './defaultLayers';
import LabelHelper from '../helpers/labels/labelHelper';

function checkAllLayers(layers, checked = true) {
  return layers.map((l) => {
    l.checked = checked;
    if (l.children) { checkAllLayers(l.children, checked); }
    return l;
  });
}

function checkAllSelected(layers, context) {
  return layers.map((l) => {
    const keys = Object.keys(context);
    if (keys.includes(l.name)) {
      l.checked = true;
      if (l.children && context[l.name] === null) {
        checkAllLayers(l.children, true);
      } else if (l.children) {
        checkAllSelected(l.children, context[l.name]);
      }
    }
    return l;
  });
}

const getSelectedFacilitiesLayers = (selected) => {
  const defaultLayersCopy = JSON.parse(JSON.stringify(defaultLayers));

  switch (selected) {
    case 'all':
      return checkAllLayers(defaultLayersCopy, true);

    case 'none':
      return checkAllLayers(defaultLayersCopy, false);

    default:
      return selected ? checkAllSelected(defaultLayersCopy, selected) : defaultLayersCopy;
  }
};

const getDefaultFilterDimensions = ({ selected, values }) => ({
  radiusfilter: {
    type: 'radiusFilter',
    disabled: true,
    values: {
      coordinates: [],
      radius: 0,
    },
  },
  commboard: {
    type: 'multiSelect',
    disabled: true,
    values: LabelHelper.get_labels('commboard'),
  },
  nta: {
    type: 'multiSelect',
    disabled: true,
    values: LabelHelper.get_labels('nta'),
  },
  borocode: {
    type: 'multiSelect',
    disabled: true,
    values: LabelHelper.get_labels('borocode'),
  },
  censtract: {
    type: 'multiSelect',
    disabled: true,
    values: LabelHelper.get_labels('censtract'),
  },
  council: {
    type: 'multiSelect',
    disabled: true,
    values: LabelHelper.get_labels('council'),
  },
  admin_policeprecinct: {
    type: 'multiSelect',
    disabled: true,
    values: LabelHelper.get_labels('policeprecinct'),
  },
  admin_schooldistrict: {
    type: 'multiSelect',
    disabled: true,
    values: LabelHelper.get_labels('schooldistrict'),
  },
  facsubgrp: {
    type: 'facilitiesLayerSelector',
    values: values || getSelectedFacilitiesLayers(selected),
  },
  optype: {
    type: 'multiSelect',
    disabled: true,
    values: [
      {
        value: 'Public',
        label: 'Public',
      },
      {
        value: 'Non-public',
        label: 'Non-public',
      },
    ],
  },
  overabbrev: {
    type: 'multiSelect',
    disabled: true,
    values: [
      {
        value: 'Amtrak',
        label: 'Amtrak',
      },
      {
        value: 'BBPC',
        label: 'Brooklyn Bridge Park Corporation',
      },
      {
        value: 'BPL',
        label: 'Brooklyn Public Library',
      },
      {
        value: 'CUNY',
        label: 'City University of New York',
      },
      {
        value: 'FBOP',
        label: 'Federal Bureau of Prisons',
      },
      {
        value: 'HRPT',
        label: 'Hudson River Park Trust',
      },
      {
        value: 'HYDC',
        label: 'Hudson Yards Development Corporation',
      },
      {
        value: 'MTA',
        label: 'Metropolitan Transportation Authority',
      },
      {
        value: 'NYCACS',
        label: 'NYC Administration for Childrens Services',
      },
      {
        value: 'NYCBOE',
        label: 'NYC Board of Elections',
      },
      {
        value: 'NYCBSA',
        label: 'NYC Board of Standards and Appeals',
      },
      {
        value: 'NYCBP-BX',
        label: 'NYC Borough President - Bronx',
      },
      {
        value: 'NYCBP-BK',
        label: 'NYC Borough President - Brooklyn',
      },
      {
        value: 'NYCBP-MN',
        label: 'NYC Borough President - Manhattan',
      },
      {
        value: 'NYCBP-QN',
        label: 'NYC Borough President - Queens',
      },
      {
        value: 'NYCBP-SI',
        label: 'NYC Borough President - Staten Island',
      },
      {
        value: 'NYCBIC',
        label: 'NYC Business Integrity Commission',
      },
      {
        value: 'NYCCFB',
        label: 'NYC Campaign Finance Board',
      },
      {
        value: 'NYCCEO',
        label: 'NYC Center for Economic Opportunity',
      },
      {
        value: 'NYCCSC',
        label: 'NYC Civil Service Commission',
      },
      {
        value: 'NYCCCRB',
        label: 'NYC Civilian Complaint Review Board',
      },
      {
        value: 'NYCCCHR',
        label: 'NYC Commission on Human Rights',
      },
      {
        value: 'NYCCB',
        label: 'NYC Community Boards',
      },
      {
        value: 'NYCCOIB',
        label: 'NYC Conflict of Interest Board',
      },
      {
        value: 'NYCCOUNCIL',
        label: 'NYC City Council',
      },
      {
        value: 'NYCDFTA',
        label: 'NYC Department for the Aging',
      },
      {
        value: 'NYCDOB',
        label: 'NYC Department of Buildings',
      },
      {
        value: 'NYCDCP',
        label: 'NYC Department of City Planning',
      },
      {
        value: 'NYCDCAS',
        label: 'NYC Department of Citywide Administrative Services',
      },
      {
        value: 'NYCDCA',
        label: 'NYC Department of Consumer Affairs',
      },
      {
        value: 'NYCDOC',
        label: 'NYC Department of Correction',
      },
      {
        value: 'NYCDCLA',
        label: 'NYC Department of Cultural Affairs',
      },
      {
        value: 'NYCDDC',
        label: 'NYC Department of Design and Construction',
      },
      {
        value: 'NYCDOE/SCA',
        label: 'NYC Department of Education/School Construction Authority',
      },
      {
        value: 'NYCDEP',
        label: 'NYC Department of Environmental Protection',
      },
      {
        value: 'NYCDOF',
        label: 'NYC Department of Finance',
      },
      {
        value: 'NYCDOHMH',
        label: 'NYC Department of Health and Mental Hygiene',
      },
      {
        value: 'NYCDHS',
        label: 'NYC Department of Homeless Services',
      },
      {
        value: 'NYCHPD',
        label: 'NYC Department of Housing Preservation and Development',
      },
      {
        value: 'NYCDOITT',
        label: 'NYC Department of Information Technology and Telecommunications',
      },
      {
        value: 'NYCDOI',
        label: 'NYC Department of Investigation',
      },
      {
        value: 'NYCDPR',
        label: 'NYC Department of Parks and Recreation',
      },
      {
        value: 'NYCDOP',
        label: 'NYC Department of Probation',
      },
      {
        value: 'NYCDORIS',
        label: 'NYC Department of Records and Information Services',
      },
      {
        value: 'NYCDSNY',
        label: 'NYC Department of Sanitation',
      },
      {
        value: 'NYCEDC',
        label: 'NYC Economic Development Corporation',
      },
      {
        value: 'NYCDOT',
        label: 'NYC Department of Transportation',
      },
      {
        value: 'NYCDYCD',
        label: 'NYC Department of Youth and Community Development',
      },
      {
        value: 'NYCDA-BX',
        label: 'NYC District Attorney - Bronx ',
      },
      {
        value: 'NYCDA-BK',
        label: 'NYC District Attorney - Brooklyn',
      },
      {
        value: 'NYCDA-MN',
        label: 'NYC District Attorney - Manhattan',
      },
      {
        value: 'NYCDA-SNP',
        label: 'NYC District Attorney - Office Special Narcotics',
      },
      {
        value: 'NYCDA-QN',
        label: 'NYC District Attorney - Queens',
      },
      {
        value: 'NYCDA-SI',
        label: 'NYC District Attorney - Staten Island',
      },
      {
        value: 'NYCEDC',
        label: 'NYC Economic Development Corporation',
      },
      {
        value: 'NYCFDNY',
        label: 'NYC Fire Department',
      },
      {
        value: 'NYCHHC',
        label: 'NYC Health and Hospitals Corporation',
      },
      {
        value: 'NYCHA',
        label: 'New York City Housing Authority',
      },
      {
        value: 'NYCHRA/DSS',
        label: 'NYC Human Resources Administration/Department of Social Services',
      },
      {
        value: 'NYCIBO',
        label: 'NYC Independent Budget Office',
      },
      {
        value: 'NYCLPC',
        label: 'NYC Landmarks Preservation Commission',
      },
      {
        value: 'NYCLAW',
        label: 'NYC Law Department',
      },
      {
        value: 'NYCOATH',
        label: 'NYC Office of Administrative Trials and Hearings',
      },
      {
        value: 'NYCOCB',
        label: 'NYC Office of Collective Bargaining',
      },
      {
        value: 'NYCOCA',
        label: 'NYC Office of Court Administration',
      },
      {
        value: 'NYCEM',
        label: 'NYC Office of Emergency Management',
      },
      {
        value: 'NYCOLR',
        label: 'NYC Office of Labor Relations',
      },
      {
        value: 'NYCOMB',
        label: 'NYC Office of Management and Budget',
      },
      {
        value: 'NYCOPA',
        label: 'NYC Office of Payroll Administration',
      },
      {
        value: 'NYCPA',
        label: 'NYC Office of Public Advocate',
      },
      {
        value: 'NYCACT',
        label: 'NYC Office of the Actuary',
      },
      {
        value: 'NYCCLERK',
        label: 'NYC Office of the City Clerk',
      },
      {
        value: 'NYCCOMP',
        label: 'NYC Office of the Comptroller',
      },
      {
        value: 'NYCMO',
        label: 'NYC Office of the Mayor',
      },
      {
        value: 'NYCOCME',
        label: 'NYC Office of the Medical Examiner',
      },
      {
        value: 'NYPD',
        label: 'NYC Police Department',
      },
      {
        value: 'NYCPA-BX',
        label: 'NYC Public Administrators Office - Bronx',
      },
      {
        value: 'NYCPA-BK',
        label: 'NYC Public Administrators Office - Brooklyn',
      },
      {
        value: 'NYCPA-MN',
        label: 'NYC Public Administrators Office - Manhattan',
      },
      {
        value: 'NYCPA-QN',
        label: 'NYC Public Administrators Office - Queens',
      },
      {
        value: 'NYCPA-SI',
        label: 'NYC Public Administrators Office - Staten Island',
      },
      {
        value: 'NYCTAXCM',
        label: 'NYC Tax Commission',
      },
      {
        value: 'NYCTLC',
        label: 'NYC Taxi and Limousine Commission',
      },
      {
        value: 'NYC-Unknown',
        label: 'NYC Unknown',
      },
      {
        value: 'NYPD',
        label: 'New York Police Department',
      },
      {
        value: 'NYPL',
        label: 'New York Public Library',
      },
      {
        value: 'NYSDOCCS',
        label: 'NYS Department of Corrections and Community Supervision',
      },
      {
        value: 'NYSDEC',
        label: 'NYS Department of Environmental Conservation',
      },
      {
        value: 'NYSDOH',
        label: 'NYS Department of Health',
      },
      {
        value: 'NYSDOT',
        label: 'NYS Department of Transportation',
      },
      {
        value: 'NYSED',
        label: 'NYS Education Department',
      },
      {
        value: 'NYSOPWDD',
        label: 'NYS Office for People With Developmental Disabilities',
      },
      {
        value: 'NYSOASAS',
        label: 'NYS Office of Alcoholism and Substance Abuse Services',
      },
      {
        value: 'NYSOCFS',
        label: 'NYS Office of Children and Family Services',
      },
      {
        value: 'NYSOMH',
        label: 'NYS Office of Mental Health',
      },
      {
        value: 'NYSOPRHP',
        label: 'NYS Office of Parks, Recreation and Historic Preservation',
      },
      {
        value: 'NYCOURTS',
        label: 'NYS Unified Court System',
      },
      {
        value: 'Non-public',
        label: 'Non-public',
      },
      {
        value: 'PANYNJ',
        label: 'Port Authority of New York and New Jersey',
      },
      {
        value: 'QPL',
        label: 'Queens Public Library',
      },
      {
        value: 'RIOC',
        label: 'Roosevelt Island Operating Corporation',
      },
      {
        value: 'TGI',
        label: 'Trust for Governors Island',
      },
      {
        value: 'USCG',
        label: 'US Coast Guard',
      },
      {
        value: 'USCOURTS',
        label: 'US Courts',
      },
      {
        value: 'USDOT',
        label: 'US Department of Transportation',
      },
      {
        value: 'USNPS',
        label: 'National Park Service',
      },
    ],
  },
});

export default getDefaultFilterDimensions;
