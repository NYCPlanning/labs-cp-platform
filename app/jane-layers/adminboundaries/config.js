const paint = {
  lines: {
    'line-color': '#717171',
    'line-opacity': 0.7,
    'line-width': {
      stops: [
        [9, 1],
        [14, 4],
      ],
    },
  },
  labels: {
    'text-color': '#626262',
    'text-halo-color': '#FFFFFF',
    'text-halo-width': 2,
    'text-halo-blur': 2,
  },
};

const layout = {
  lines: {
    'line-join': 'round',
    'line-cap': 'round',
  },
};

const mapLayers = {
  ntaboundaries: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  ntaboundaries_labels: {
    type: 'symbol',
    minzoom: 13,
    paint: paint.labels,
    layout: {
      'text-field': '{ntaname}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  censustracts: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  censustracts_labels: {
    type: 'symbol',
    minzoom: 13,
    paint: paint.labels,
    layout: {
      'text-field': '{ct_2010}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 9],
          [14, 14],
        ],
      },
    },
  },
  puma: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  puma_labels: {
    type: 'symbol',
    minzoom: 13,
    paint: paint.labels,
    layout: {
      'text-field': '{puma}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  taz: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  taz_labels: {
    type: 'symbol',
    minzoom: 13,
    paint: paint.labels,
    layout: {
      'text-field': '{TAZCE10}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 9],
          [14, 14],
        ],
      },
    },
  },
  cd: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  cd_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{displaynam}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  schooldistricts: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  schooldistricts_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{SchoolDist}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  boroughboundaries: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  boroughboundaries_labels: {
    minzoom: 10,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{BoroName}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  citycouncildistricts: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  citycouncildistricts_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{CounDist}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  firebattalions: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  firebattalions_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{FireBN}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  firecompanies: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  firecompanies_labels: {
    minzoom: 12,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{FireCoNum}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  firedivisions: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  firedivisions_labels: {
    minzoom: 10,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{FireDiv}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  municipalcourtdistricts: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  municipalcourtdistricts_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{MuniCourt}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  policeprecincts: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  policeprecincts_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{Precinct}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  stateassemblydistricts: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  stateassemblydistricts_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{AssemDist}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  statesenatedistricts: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  statesenatedistricts_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{StSenDist}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
  uscongressionaldistricts: {
    type: 'line',
    paint: paint.lines,
    layout: layout.lines,
  },
  uscongressionaldistricts_labels: {
    minzoom: 11,
    type: 'symbol',
    paint: paint.labels,
    layout: {
      'text-field': '{CongDist}',
      'text-font': ['Open Sans Semibold', 'Arial Unicode MS Bold'],
      'text-size': {
        stops: [
          [11, 12],
          [14, 16],
        ],
      },
    },
  },
};

export default mapLayers;
