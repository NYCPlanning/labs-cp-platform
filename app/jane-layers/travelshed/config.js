const config = {
  sources: [
    {
      id: 'travelshed',
      type: 'geojson',
    },
  ],
  mapLayers: [
    {
      id: 'travelshed',
      type: 'fill',
      source: 'travelshed',
      paint: {
        'fill-color': 'steelblue',
        'fill-opacity': 0.3,
        'fill-outline-color': 'gray',
      },
    },
  ],
};

export default config;
