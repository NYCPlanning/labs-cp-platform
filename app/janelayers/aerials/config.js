import appConfig from '../../helpers/appConfig';

const config = {
  sources: [
    {
      id: 'nyaerials',
      type: 'raster',
      tileSize: 256,
      tiles: `//${appConfig.api_domain}/tiles/doitt/tms/1.0.0/photo/2016/{z}/{x}/{y}.png`,
    },
  ],
  mapLayers: [
    {
      id: 'nyaerials',
      type: 'raster',
      source: 'nyaerials',
    },
  ],
};

export default config;
