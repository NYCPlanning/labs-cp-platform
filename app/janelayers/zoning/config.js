const appConfig = {
  carto_user: 'cpp',
  carto_domain: 'cartoprod.capitalplanning.nyc',
};

const paint = {
  labels: {
    'text-color': '#626262',
    'text-halo-color': '#FFFFFF',
    'text-halo-width': 2,
    'text-halo-blur': 2,
  },
  co_labels: {
    'text-color': 'rgba(255, 0, 0, 1)',
    'text-halo-color': '#FFFFFF',
    'text-halo-width': 2,
    'text-halo-blur': 2,
  },
};

const config = {
  zd: {
    sources: [
      {
        id: 'zd',
        type: 'cartovector',
        options: {
          carto_user: appConfig.carto_user,
          carto_domain: appConfig.carto_domain,
          sql: ['SELECT *, LEFT(zonedist, 2) as primaryzone FROM support_zoning_zd'],
        },
      },
    ],
    mapLayers: [
      {
        id: 'zd',
        type: 'fill',
        source: 'zd',
        'source-layer': 'layer0',
        paint: {
          'fill-color': {
            property: 'primaryzone',
            type: 'categorical',
            stops: [
              ['BP', '#EEEEEE'],
              ['C1', '#FDBDBB'],
              ['C2', '#FDBDBB'],
              ['C3', '#FDBDBB'],
              ['C4', '#FDBDBB'],
              ['C5', '#FDBDBB'],
              ['C6', '#FDBDBB'],
              ['C7', '#FDBDBB'],
              ['C8', '#FDBDBB'],
              ['M1', '#B7D6FD'],
              ['M2', '#EDB7FD'],
              ['M3', '#EDB7FD'],
              ['PA', '#E7FDDC'],
              ['R1', '#FDFDDC'],
              ['R2', '#FDFDDC'],
              ['R3', '#FDFDDC'],
              ['R4', '#FDFDDC'],
              ['R5', '#FDFDDC'],
              ['R6', '#FDE7BD'],
              ['R7', '#FDE7BD'],
              ['R8', '#FDE7BD'],
              ['R9', '#FDE7BD'],
            ],
          },
          'fill-opacity': 0.6,
          'fill-antialias': true,
          'fill-outline-color': 'rgba(0, 0, 0, 1)',
        },
      },
      {
        id: 'zd_labels',
        source: 'zd',
        type: 'symbol',
        'source-layer': 'layer0',
        paint: paint.labels,
        layout: {
          'symbol-placement': 'point',
          'text-field': '{zonedist}',
        },
        minzoom: 14,
      },
    ],
  },
  co: {
    sources: [
      {
        id: 'co',
        type: 'cartovector',
        options: {
          carto_user: appConfig.carto_user,
          carto_domain: appConfig.carto_domain,
          sql: ['SELECT * FROM support_zoning_co'],
        },
      },
    ],
    mapLayers: [
      {
        id: 'co',
        type: 'fill',
        source: 'co',
        'source-layer': 'layer0',
        paint: {
          'fill-opacity': 1,
          'fill-color': 'rgba(158, 0, 0, 0)',
          'fill-antialias': true,
          'fill-outline-color': 'rgba(255, 0, 0, 1)',
        },
      },
      {
        id: 'co_labels',
        source: 'co',
        type: 'symbol',
        'source-layer': 'layer0',
        paint: paint.co_labels,
        layout: {
          'symbol-placement': 'point',
          'text-field': '{overlay}',
        },
        minzoom: 14,
      },
    ],
  },
};

export default config;
