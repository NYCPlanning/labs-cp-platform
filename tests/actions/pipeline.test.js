import * as actions from '../../app/actions/pipeline';
import { defaultFilterDimensions } from '../../app/pipeline/config';

describe('actions/pipeline', () => {
  test('fetchDetails', () =>
    expect(actions.fetchDetails('cartodbId')).toMatchSnapshot());

  test('fetchTotalCount', () =>
    expect(actions.fetchTotalCount()).toMatchSnapshot());

  test('fetchSelectedCount', () =>
    expect(actions.fetchSelectedCount(defaultFilterDimensions)).toMatchSnapshot());

  test('setSelectedFeatures', () =>
    expect(actions.setSelectedFeatures([])).toMatchSnapshot());

  test('setFilterDimension', () =>
    expect(actions.setFilterDimension('dimension', 'values')).toMatchSnapshot());

  test('setSymbology', () =>
    expect(actions.setSymbology('symbologyDimension')).toMatchSnapshot());

  test('resetFilters', () =>
    expect(actions.resetFilters()).toMatchSnapshot());
});
