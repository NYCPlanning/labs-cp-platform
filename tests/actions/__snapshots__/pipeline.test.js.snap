// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/pipeline fetchDetails 1`] = `
Object {
  "payload": Object {
    "nextType": Object {
      "FAILURE": "FETCH_PIPELINE_DETAILS_FAILURE",
      "PENDING": "FETCH_PIPELINE_DETAILS_PENDING",
      "SUCCESS": "FETCH_PIPELINE_DETAILS_SUCCESS",
    },
    "requestFormat": "geojson",
    "sql": "SELECT * FROM pipeline_projects_prod WHERE cartodb_id = 'cartodbId'",
  },
  "type": "CARTO_REQUEST",
}
`;

exports[`actions/pipeline fetchSelectedCount 1`] = `
Object {
  "payload": Object {
    "nextType": Object {
      "FAILURE": "FETCH_PIPELINE_SELECTED_COUNT_FAILURE",
      "PENDING": "FETCH_PIPELINE_SELECTED_COUNT_PENDING",
      "SUCCESS": "FETCH_PIPELINE_SELECTED_COUNT_SUCCESS",
    },
    "requestFormat": "json",
    "sql": "SELECT count(*) FROM (SELECT cartodb_id, the_geom_webmercator, dcp_pipeline_status, dcp_permit_type, dcp_units_use_map, dob_permit_address FROM pipeline_projects_prod WHERE (dcp_pipeline_status = 'Complete' OR dcp_pipeline_status = 'Partial complete' OR dcp_pipeline_status = 'Permit issued' OR dcp_pipeline_status = 'Application filed' OR dcp_pipeline_status = 'Demolition (complete)') AND (dcp_permit_type = 'New Building' OR dcp_permit_type = 'Alteration' OR dcp_permit_type = 'Demolition') AND (dcp_development_type = 'Residential' OR dcp_development_type = 'Non-residential') AND (dcp_units_use_map >= '-1445' AND dcp_units_use_map <= '1669')) a",
  },
  "type": "CARTO_REQUEST",
}
`;

exports[`actions/pipeline fetchTotalCount 1`] = `
Object {
  "payload": Object {
    "nextType": Object {
      "FAILURE": "FETCH_PIPELINE_TOTAL_COUNT_FAILURE",
      "PENDING": "FETCH_PIPELINE_TOTAL_COUNT_PENDING",
      "SUCCESS": "FETCH_PIPELINE_TOTAL_COUNT_SUCCESS",
    },
    "requestFormat": "json",
    "sql": "SELECT COUNT(*) FROM pipeline_projects_prod",
  },
  "type": "CARTO_REQUEST",
}
`;

exports[`actions/pipeline resetFilters 1`] = `
Object {
  "type": "RESET_PIPELINE_FILTERS",
}
`;

exports[`actions/pipeline setFilterDimension 1`] = `
Object {
  "payload": Object {
    "filterDimension": "dimension",
    "values": "values",
  },
  "type": "SET_PIPELINE_FILTER_DIMENSION",
}
`;

exports[`actions/pipeline setSelectedFeatures 1`] = `
Object {
  "payload": Object {
    "selectedFeatures": Array [],
  },
  "type": "SET_SELECTED_PIPELINE_FEATURES",
}
`;

exports[`actions/pipeline setSymbology 1`] = `
Object {
  "payload": Object {
    "symbologyDimension": "symbologyDimension",
  },
  "type": "SET_PIPELINE_SYMBOLOGY",
}
`;
